### PR TITLE
Support Spark 2.3.1, TF 1.10.0 and drop Spark 2.1/2.2 (and hence Scala 2.10, Java 7)

### PR DESCRIPTION
**Pull request by mengxr**
* Drop support for Spark 2.1 and 2.2 and hence scala 2.10 and java 7
* Update TF to 1.10 release
* Remove nix files, which are not used
* Update README

We will support Spark 2.4 once RC is released.

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -1,10 +1,5 @@
...
-
 object Dependencies {
   // The spark version
-  val targetSparkVersion = sys.props.getOrElse("spark.version", "2.3.0")
-  val targetTensorFlowVersion = "1.6.0"
-
-}
\ No newline at end of file
+  val targetSparkVersion = sys.props.getOrElse("spark.version", "2.3.1")
+  val targetTensorFlowVersion = "1.10.0-rc1"
```
Comment (mengxr) [inline project/Dependencies.scala]: TensorFlow hasn't released 1.10 for its Java API yet.

**Comment by thunterdb** [comment]
@mengxr you also need to update the proto java stubs, otherwise the protos written by tensorflow 1.10 will not be understood. This is what this PR is doing: https://github.com/databricks/tensorframes/pull/149
Or, if they the java stubs are now built into the official java interface, they should be removed from tensorframes.

**Comment by mengxr** [comment]
@thunterdb Thanks for the note! Are you aware of other TODOs before the release?

**Comment by thunterdb** [comment]
@mengxr the proto update is the only non-standard step compared to building other spark packages. I am not aware of any outstanding request.

Be aware that in #149, the device placement is logged:
https://github.com/databricks/tensorframes/pull/149/files#diff-8d473331e016e149ade1b3490362d3d2R156

This is useful for checking if GPUs are being used, but it may add too much output to logging. The changes to that file are not relevant as long as we do not have an official GPU version of tensorframes.

**Pull request merged**